### PR TITLE
Added support for running evaluators locally

### PR DESCRIPTION
**Pull request by andrius-k**
All evaluators are now fully supported to be executed locally. This requires that users set their OpenAI API key:
``` bash
export OPENAI_API_KEY=your-openai-api-key
```

Lynxius API key is also required to store the evaluator output in the Lynxius online platform:
``` bash
export LYNXIUS_API_KEY=your-lynxius-api-key
```

Usage example:
```python
from lynxius.evals.local.bert_score import BertScoreLocal

bert_score_local = BertScoreLocal("PR #222", level="word", presence_threshold=0.65, tags=["local"])
bert_score_local.add_trace(reference="reference here", output="actual output here", context=[...])

# Evaluate locally
bert_score_local.evaluate()

# Store the results in the Lynxius platform
client.store(bert_score_local)
```

Fully working examples of each evaluator are available in the dedicated notebook: `tutorials/GettingStarted.ipynb`.

This PR adds additional dependencies to the `requirements.txt` file.

### CODE REVIEW

**Review by dmpiergiacomo — APPROVED** [review]
Great work! Left a few comments.

**Comment by andrius-k** [comment]
I changed the error messages, nice catch! Also the dataset retrieval code will now raise an appropriate exception if the HTTP request didn't respond with a `2xx` status.

File: lynxius/client.py [hunk 1]
```diff
@@ -59,6 +60,29 @@ def evaluate(self, eval: Evaluator) -> str | None:
...
+            print("Error:", response.status_code, response.text)
+            return None
+
+    def store(self, eval: EvaluatorLocal) -> str | None:
+        """
+        Stores the locally executed evaluator on the Lynxius platform.
+        Returns an eval run ID.
+        """
+
+        if not isinstance(eval, EvaluatorLocal):
```
[the remark this answers is not in the document]
Comment (dmpiergiacomo) [reply lynxius/client.py:83]: You are right, this is correct. I missed the `not` in the line. I gest I got confused by line 63 `if isinstance(eval, EvaluatorLocal):` that does not use the `not` and I was expecting the same construct on line 83.

All good :+1:

**Comment by dmpiergiacomo** [comment]
Thanks and great addition about the HTTP error status!

**Pull request merged**